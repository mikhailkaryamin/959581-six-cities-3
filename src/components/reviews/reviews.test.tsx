import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import {
  configure,
  shallow
} from 'enzyme';
import Reviews from './reviews';

configure({
  adapter: new Adapter(),
});

const comments = [
  {
    id: 1,
    user: {
      id: 1,
      isPro: true,
      name: `Mi`,
      avatarUrl: `img/avatar-max.jpg`
    },
    rating: 2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Curabitur ornare nunc a blandit ultricies.
    Integer a urna in nunc congue efficitur.`,
    date: `April 2010`
  }
];
const isAuth = true;
const onCommentSubmit = jest.fn();
const responseStatus = null;
const countComments = 3;

test(`Render reviews`, () => {
  const tree = shallow(
      <Reviews
        comments={comments}
        countComments={countComments}
        isAuth={isAuth}
        onCommentSubmit={onCommentSubmit}
        responseStatus={responseStatus}
      />
  ).html();

  expect(tree).toMatchSnapshot();
});
