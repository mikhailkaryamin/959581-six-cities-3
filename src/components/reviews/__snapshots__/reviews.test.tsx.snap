// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render reviews 1`] = `
"<section class=\\"property__reviews reviews\\"><h2 class=\\"reviews__title\\">Reviews Â·<span class=\\"reviews__amount\\">3</span></h2><ul class=\\"reviews__list\\"><li class=\\"reviews__item\\"><div class=\\"reviews__user user\\"><div class=\\"reviews__avatar-wrapper user__avatar-wrapper\\"><img class=\\"reviews__avatar user__avatar\\" src=\\"img/avatar-max.jpg\\" width=\\"54\\" height=\\"54\\" alt=\\"Reviews avatar\\"/></div><span class=\\"reviews__user-name\\">Mi</span></div><div class=\\"reviews__info\\"><div class=\\"reviews__rating rating\\"><div class=\\"reviews__stars rating__stars\\"><span style=\\"width:40%\\"></span><span class=\\"visually-hidden\\">Rating</span></div></div><p class=\\"reviews__text\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Curabitur ornare nunc a blandit ultricies.
    Integer a urna in nunc congue efficitur.</p><time class=\\"reviews__time\\" dateTime=\\"April 2010\\">April 2010</time></div></li></ul><form class=\\"reviews__form form\\" action=\\"#\\" method=\\"post\\"><label class=\\"reviews__label form__label\\" for=\\"review\\">Your review</label><div class=\\"reviews__rating-form form__rating\\"><input type=\\"radio\\" class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"5\\" id=\\"5-stars\\"/><label for=\\"5-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input type=\\"radio\\" class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"4\\" id=\\"4-stars\\"/><label for=\\"4-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input type=\\"radio\\" class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"3\\" id=\\"3-stars\\"/><label for=\\"3-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input type=\\"radio\\" class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"2\\" id=\\"2-stars\\"/><label for=\\"2-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label><input type=\\"radio\\" class=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"1\\" id=\\"1-stars\\"/><label for=\\"1-stars\\" class=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\"><svg class=\\"form__star-image\\" width=\\"37\\" height=\\"33\\"><use xlink:href=\\"#icon-star\\"></use></svg></label></div><textarea class=\\"reviews__textarea form__textarea\\" id=\\"review\\" name=\\"review\\" placeholder=\\"Tell how was your stay, what you like and what can be improved\\"></textarea><div class=\\"reviews__button-wrapper\\"><p class=\\"reviews__help\\">To submit review please make sure to set<span class=\\"reviews__star\\">rating</span>and describe your stay with at least<b class=\\"reviews__text-amount\\">50 characters</b>.</p><button class=\\"reviews__submit form__submit button\\" type=\\"submit\\">Submit</button></div></form></section>"
`;
